### PR TITLE
Change nav behavior on click; add submenu to mobile nav

### PR DESCRIPTION
**Pull request by JoeShep**
This change eliminates taking the user to `/places` or `/bookings` when clicking on the Destinations or Bookings nav links. In mobile views, the click now opens a submenu. A call to action btn is also added to Bookings dropdown. 

Note that this PR is for the functionality changes only. There were no new designs to go by. I plan to review the changes with a designer after merging and will update styling after that review.

Needs https://github.com/lonelyplanet/destinations-next/pull/471 to make Bookings CTA appear

screen shot 2015-11-02 at 10 36 05 am

### CODE REVIEW

File: src/components/navigation/navigation_component.js [hunk 1]
```diff
@@ -25,6 +28,10 @@ class NavigationComponent extends Component {
     this.name = "navigation";
     this.$mobileNavigation = this.$el.find(".mobile-navigation").detach();
     this.$mobileNavigation.on("click", ".js-close", this._clickNav.bind(this));
+    this.$mobileNavigation.on("click", ".js-nav-item", this._handleClick.bind(this));
+
+    console.log("Element", this.$el);
```
Comment (jcreamer898) [inline src/components/navigation/navigation_component.js]: oops console log in here still

File: src/components/navigation/navigation_component.js [hunk 2]
```diff
@@ -40,27 +47,45 @@ class NavigationComponent extends Component {
...
    * @return {[type]} [description]
    */
-  handleHover() {
-    let hideTimer, showTimer;
 
-    this.$el.on("mouseenter", ".navigation__item", (event) => {
-      clearTimeout(hideTimer);
+  _handleClick(e) {
+    let target = e.currentTarget;
+    if (target.text === "Shop")
```
Comment (jcreamer898) [inline src/components/navigation/navigation_component.js]: plz I can haz `{}`

**Comment by JoeShep** [comment]
Changes made, @jcreamer898

**Comment by jcreamer898** [comment]
cool

**Pull request merged**